src/timeseriesflattener/text_embedding_functions.py

```diff
@@ -0,0 +1,95 @@
+"""Functions for embedding text data"""
+from collections.abc import Callable
+from typing import Optional
+
+import pandas as pd
+import torch
+from sentence_transformers import SentenceTransformer
+from sklearn.base import TransformerMixin
+from transformers import pipeline
+
+
+def embed_text_column(
+    df: pd.DataFrame,
+    text_col_name: str,
+    embedding_fn: Callable,
+    embedding_fn_kwargs: Optional[dict] = None,
+) -> pd.DataFrame:
+    """Embeds text values using the embedding_fn and optionally reduces the
+    dimensionality using dim_reduction_fn. Stores the embedding as a multi-
+    index column called 'value'.
+
```

Comment:
You removed the dimensionality reduction functionality completely? :-)

Reply:
Yes, after giving it some more though, it makes much more sense to handle during model training. It requires a model that's fit to the embedding, so it's way more convenient, fast, (and good hyperparam) to do afterwards

Reply:
Ah, yes, it's actually a post-split preprocessing step! Good catch!